src/Application/CommandHandlers/billingParty/UpdateBillingPartyHandler.cs

```diff
@@ -0,0 +1,80 @@
+    private readonly IUniqueBillingPartyVatNumberChecker _billingPartyBillingPartyVatNumberChecker;
+    private readonly IUniqueBillingPartyNameChecker _billingPartyNameChecker;
+
+
+    public UpdateBillingPartyHandler(IBillingPartyRepository billingPartyRepository, IUnitOfWork unitOfWork, IUniqueBillingPartyEmailChecker billingPartyEmailChecker, IUniqueBillingPartyVatNumberChecker billingPartyBillingPartyVatNumberChecker, IUniqueBillingPartyNameChecker billingPartyNameChecker) {
+        _billingPartyRepository = billingPartyRepository;
+        _unitOfWork = unitOfWork;
+        _billingPartyEmailChecker = billingPartyEmailChecker;
+        _billingPartyBillingPartyVatNumberChecker = billingPartyBillingPartyVatNumberChecker;
+        _billingPartyNameChecker = billingPartyNameChecker;
+    }
+
+    public async Task Handle(UpdateBillingPartyCommand.Request request, CancellationToken cancellationToken) {
+
+        BillingPartyEntity partyEntity = new BillingPartyEntity() {
+            Id = Guid.NewGuid(),
+            Name = request.Name,
+            Address = request.Address,
+            Email = request.Email,
+            PhoneNumber = request.PhoneNumber,
+            VatNumber = request.VatNumber
+        };
+
+        bool tryParse = Guid.TryParse(request.Id, out Guid guid);
+        if (!tryParse) {
+            throw new DomainValidationException("Id", ErrorCode.BadRequest, ErrorMessages.IdInvalid(request.Id));
+        }
+
+        BillingPartyEntity? billingPartyEntity = await _billingPartyRepository.GetByIdAsync(guid);
+        if (billingPartyEntity is null) {
+            throw new DomainValidationException("Id", ErrorCode.NotFound, ErrorMessages.BillingPartyNotFound(guid));
+        }
+
+        if (!string.IsNullOrEmpty(partyEntity.Name)) {
+            bool isUnique = await _billingPartyNameChecker.IsUniqueAsync(partyEntity.Name, guid);
+            if (!isUnique) {
+                throw new DomainValidationException("Email", ErrorCode.Conflict,
+                    ErrorMessages.BillingPartyNameAlreadyExists(partyEntity.Name));
+            }
+        }
+
+        if (!string.IsNullOrEmpty(partyEntity.Email)) {
+            bool isUnique = await _billingPartyEmailChecker.IsUniqueAsync(partyEntity.Email, guid);
+            if (!isUnique) {
+                throw new DomainValidationException("Email", ErrorCode.Conflict,
+                    ErrorMessages.BillingPartyEmailAlreadyExists(partyEntity.Email));
+            }
+        }
+
+        if (!string.IsNullOrEmpty(partyEntity.VatNumber)) {
+            bool isUnique = await _billingPartyBillingPartyVatNumberChecker.IsUniqueAsync(partyEntity.VatNumber, guid);
+            if (!isUnique) {
+                throw new DomainValidationException("Email", ErrorCode.Conflict,
+                    ErrorMessages.BillingPartyVatNumberAlreadyExists(partyEntity.VatNumber));
+            }
+        }
+
+        billingPartyEntity.PhoneNumber = request.PhoneNumber;
```

Comment:
I was thinking of instead of directly saving changes do we want to check if the field has changed ?

Reply:
I mean if the field did not change, we are not setting it to anything different either.

For example if it had some value as "test" and then it did not change , it sets that to "test" again. 
So, i do not see any problem here.